src/XRP/model/xrp-transaction.ts

```diff
@@ -6,31 +6,42 @@ import XRPTransactionType from './xrp-transaction-type'
 import XRPPayment from './xrp-payment'
 import XRPMemo from './xrp-memo'
 import { GetTransactionResponse } from '../Generated/web/org/xrpl/rpc/v1/get_transaction_pb'
+import XRPLNetwork from '../../Common/xrpl-network'
 
 /*
  * A transaction on the XRP Ledger.
  *
  * @see: https://xrpl.org/transaction-formats.html
  */
-// TODO(amiecorso): Modify this object to use X-Address format.
 export default class XRPTransaction {
   /**
    * Constructs an XRPTransaction from a Transaction.
    *
    * @param transaction a Transaction (protobuf object) whose field values will be used
    *                    to construct an XRPTransaction
+   * @param xrplNetwork The XRPL network from which this object was retrieved.
    * @returns an XRPTransaction with its fields set via the analogous protobuf fields.
    * @see https://github.com/ripple/rippled/blob/develop/src/ripple/proto/org/xrpl/rpc/v1/transaction.proto#L13
    */
   public static from(
     getTransactionResponse: GetTransactionResponse,
+    xrplNetwork: XRPLNetwork,
   ): XRPTransaction | undefined {
     const transaction = getTransactionResponse.getTransaction()
     if (!transaction) {
       return undefined
     }
 
     const account = transaction.getAccount()?.getValue()?.getAddress()
+    if (!account) {
+      return undefined
+    }
+
+    const accountXAddress = Utils.encodeXAddress(
+      account,
+      undefined,
```

Comment:
shouldn't we pass sourceAddress here?

Reply:
EDIT: I see what's happening. 

I think there should just be a `sourceXAddress` which is the union of `account` and `sourceTag`. Instead of having two source X-Addresses. 

[The doc for `sourceTag`](https://xrpl.org/transaction-common-fields.html) says:
"""
Arbitrary integer used to identify the reason for this payment, or a sender on whose behalf this transaction is made. Conventionally, a refund should specify the initial payment's SourceTag as the refund payment's DestinationTag.
"""

So you could imagine that you want to reverse a payment, you'd just use:
```
const paymentToReverse = ...
const destinationForReverse  = paymentToReverse.sourceXAddress
```

And things just work. Is there a reason you think you'd need both addresses?